class6/StringCompressor.java

```diff
@@ -1,27 +1,62 @@
+package IceBreakers;
 
-/**
- * Created by Icebreakers on 3/5/17.
- * Implement a method to perform basic string compression using the counts
- of repeated characters.
- For example, the string aabcccccaaa would become a2blc5a3.
- If the "compressed" string would not become smaller than the original
- string, your method should return the original string.
- *
- */
-public class StringCompressor {
+/*
+Input :
+Input :a
+Output :a
+Input :aabcccccaaa
+Output :a2b1c5a3
+Input :abcdedf
+Output :abcdedf
+*/
 
-    private static String compressedString(String s){
-
-    }
-
-    public static void testBench(){
-        //null case
-        // 1 character string
-        // string with multiple characters case
-        // compressed string > original string case
-    }
-    public static void main(String args[]){
-        testBench();
-    }
+public class StringCompression {
+	String str;
+	StringCompression(String str) {
+		System.out.println("Input :" + str);
+		this.str = str;
+	}
+	StringCompression(){
+		
+	}
+	
+	void compress() {
+		if(str == null || str.isEmpty()) return;
+		String res = "";
+		res += str.charAt(0);
+	    char prev = str.charAt(0);
+	    int cnt = 1;
+	    char curr;
+		for(int i=1;i<str.length();i++) {
```

Comment:
Good use of two pointer strategy. But the drawback is you need to be mindful of re-assigning them properly.